src/shared/networkrequest.cpp

```diff
@@ -200,8 +200,7 @@ void NetworkRequest::replyFinished() {
   logger.debug() << "Network reply received - status:" << status
                  << "- expected:" << expect;
 
-  QByteArray data = m_reply->readAll();
-  processData(m_reply->error(), m_reply->errorString(), status, data);
+  processData(m_reply->error(), m_reply->errorString(), status, m_replyData);
```

Comment:
I don't fully understand the root cause either, but I suspect that since Qt 6.3.x something has changed in HTTP errors are handled. What we observe is that the payload can be read from within the `readyRead()` signal handling, but the payload data is empty upon the `finished()` signal. It as though having the signal connected (such as directly via the `QNetworkReply::downloadProgress()`) then we must handle the data.

For whatever reason, this behavior(or bug?) doesn't manifest with a 2xx status code, so it only affects the authentication flow.

Reply:
Okay, after reading through the bug report. I have an explanation for what broke and why this fixes it.

Starting in Qt 6.3.x, some significant refactoring was done to how the `QNetworkReply` handles stream decompression in an attempt to make things more efficient and cut down on signal emissions. However, they introduced a bug that would erroneously clear the HTTP payload if an error occurs. This means that by the time we get the `QNetworkReply::finished()` signal, it's too late and our payload is gone.

However, by catching the `QIODevice::readyRead()` signal we are inserting ourselves a little earlier into the data processing cycle and grabbing the payload data *before* the `QNetworkReply` class gets a chance to trigger the bug and obliterates our payload.

Reply:
To guard against the possibility of signal ordering issues (eg: if `finished()` occurs before the final `readyRead()`) I have added an extra call to `QIODevice::readAll()` to make sure we get all the available data before handling the reply. Hopefully that should address your concerns.